mdal/api/mdal.h

```diff
@@ -94,41 +96,54 @@ MDAL_EXPORT int MDAL_M_vertexCount( MeshH mesh );
-MDAL_EXPORT void MDAL_M_vertices( MeshDataIteratorH iterator, double *coordinates );
-//! Closes mesh data iterator, frees the memory
-MDAL_EXPORT void MDAL_M_CloseMeshDataIterator( MeshDataIteratorH iterator );
-
 //! Loads dataset file. On error see MDAL_LastStatus for error type.
 //! This may effectively load whole dataset in-memory for some providers
 //! Datasets will be closed automatically on mesh destruction or memory
 //! can be freed manually with MDAL_CloseDataset if needed
 MDAL_EXPORT void MDAL_M_LoadDatasets( MeshH mesh, const char *datasetFile );
-
 //! Returns dataset groups count
 MDAL_EXPORT int MDAL_M_datasetGroupCount( MeshH mesh );
-
 //! Returns dataset group handle
 MDAL_EXPORT DatasetGroupH MDAL_M_datasetGroup( MeshH mesh, int index );
 
+///////////////////////////////////////////////////////////////////////////////////////
+/// MESH VERTICES
+///////////////////////////////////////////////////////////////////////////////////////
+
+//! Returns iterator to the mesh vertices
+//! For some formats this may effectively load all vertices in-memory unit iterator is closed
+MDAL_EXPORT MeshVerticesIteratorH MDAL_M_readVertices( MeshH mesh );
+//! Returns vertices from iterator for the mesh
+//! \param iterator mesh data iterator
+//! \param verticesCount maximum number or vertices to be written to buffer
+//! \param coordinates must be allocated to 3* verticesCount items to store x1, y1, z1, ..., xN, yN, zN coordinates
+//! \returns number of vertices written in the buffer
+MDAL_EXPORT int MDAL_M_vertices( MeshVerticesIteratorH iterator, int verticesCount, double *coordinates );
+//! Closes mesh data iterator, frees the memory
+MDAL_EXPORT void MDAL_M_CloseMeshVerticesIterator( MeshVerticesIteratorH iterator );
+
+///////////////////////////////////////////////////////////////////////////////////////
+/// MESH FACES
+///////////////////////////////////////////////////////////////////////////////////////
+
+//! Returns iterator to the mesh vertices
+//! For some formats this may effectively load all vertices in-memory unit iterator is closed
+MDAL_EXPORT MeshFacesIteratorH MDAL_M_readFaces( MeshH mesh );
+//! Returns next faces from iterator for the mesh
+//! \param iterator mesh data iterator
+//! \param facesCount maximum number or faces to be written to buffer
+//! \param faceVerticesStartIndex array must be allocated to be able to store facesCount + 1 items
+//! \param facesVertexIndicesCount number of vertex indexes buffer can store, minimum is MDAL_M_faceVerticesMaximumCount()
+//! \param facesVertexIndices writes vertex indexes for faces
+//! \returns number of faces written in the buffer
+MDAL_EXPORT int MDAL_M_faces( MeshFacesIteratorH iterator,
+                              int facesCount,
+                              int *faceVerticesStartIndex,
+                              int facesVertexIndicesCount,
+                              int *facesVertexIndices );
```

Comment:
I have trouble connecting that `facesCount` and `faceVerticesStartIndex` arguments work together as they have completely different names. And I need to read everything twice to remember which of `faceVerticesStartIndex` and `facesVertexIndices` means what. How about e.g.:
- vertexIndicesBuffer
- vertexIndicesBufferLen
- faceOffsetsBuffer
- faceOffsetsBufferLen